clients/hadoopfs/src/test/java/io/lakefs/LakeFSFileSystemTest.java

```diff
@@ -289,41 +288,191 @@ public void testAppend() throws IOException {
         fs.append(null, 0, null);
     }
 
-    /*
+    private void mockNonExistingPath(ObjectLocation objectLoc) throws ApiException {
+        when(objectsApi.statObject(objectLoc.getRepository(), objectLoc.getRef(), objectLoc.getPath()))
+                .thenThrow(new ApiException(HttpStatus.SC_NOT_FOUND, "no such file"));
+        when(objectsApi.listObjects(objectLoc.getRepository(), objectLoc.getRef(),
+                objectLoc.getPath() + Constants.SEPARATOR, "", 1, ""))
```

Comment:
AFAICT the "1" here should be an `any()` (and then, unfortunately, Mockito will probably want all the parameters to be `eq("")` and similar): *any* file listing should return nothing, no matter how many entries I ask for.

Reply:
The problem is that Mockito does not allow mixing matcher (e.g. any()) with explicit parameters..